[JENKINS-64124] Support detecting automatic build with causes

gmshake opened the pull request:
This is partial implementation to support @agabrys 's request. See https://github.com/jenkinsci/branch-api-plugin/pull/186 for full context.

Created a Jira issue to track this, https://issues.jenkins-ci.org/browse/JENKINS-64124

bitwiseman commented on src/main/java/jenkins/branch/BranchBuildStrategy.java:
```diff
@@ -160,13 +161,45 @@ public boolean isAutomaticBuild(@NonNull SCMSource source,
+     * {@link SCMHead} has been detected as created / modified.
+     * @since 2.5.7
+     */
+    @SuppressWarnings("deprecation")
```
```suggestion
```

gmshake commented:
Hi, @bitwiseman 
> We'll need some tests showing it working.

Do you mean we need UT or IT, or just test plan for this?

bitwiseman commented:
@gmshake 
I mean we need tests that pass values and exercise this new code.  Unit tests in this project at minimum.

gmshake commented:
Hi, @bitwiseman 

1. Rebased to lastest master.
2. Updated the since tag / comments

> I mean we need tests that pass values and exercise this new code. Unit tests in this project at minimum.

3. Unit tests added.

gmshake commented:
Looks the commit message need be amended.

gmshake commented:
ping @bitwiseman

jmendiara commented:
Really interested in this PR moving fordward. @bitwiseman would you mind to have a look, please?

gmshake commented:
Close as @agabrys has fixed https://github.com/jenkinsci/branch-api-plugin/pull/186 by https://github.com/jenkinsci/branch-api-plugin/pull/244 .
